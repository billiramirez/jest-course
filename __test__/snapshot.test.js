import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("It's time to the instant", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test("always is gonna fail the instant", () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });

  test("we allow an exception", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "billi ramirez obregon",
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
