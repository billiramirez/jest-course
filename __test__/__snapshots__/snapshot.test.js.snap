// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It's time to the instant Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`It's time to the instant always is gonna fail the instant 1`] = `
Object {
  "createAt": 2020-08-10T20:52:26.719Z,
  "id": 10,
}
`;

exports[`It's time to the instant we allow an exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "billi ramirez obregon",
}
`;
